repos/system_upgrade/el7toel8/actors/postgresqlcheck/libraries/library.py

```diff
@@ -0,0 +1,78 @@
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import CalledProcessError, api, run
+
+def get_postgresql_packages():
+    """ Get list of installed PostgreSQL packages """
+
+    for rpm_pkgs in api.consume(InstalledRedHatSignedRPM):
+        return [pkg for pkg in rpm_pkgs.items if pkg.name.startswith('postgresql')]
+
+def get_postgresql_extensions():
+    """ Get list of PostgreSQL extensions """
+    result = []
+    try:
+        output = run([
+            '/usr/bin/su',
+            '-l', 'postgresql'
+            '-c', 'psql -tAc \'\dx\''
+        ], split=True)['stdout']
+    except CalledProcessError:
+        api.current_logger().debug('Unable to list PostgreSQL extensions: psql failed to run')
+        return False
+
+    for entry in output:
+        entry = entry.strip()
+        if not entry:
+            continue
+        name, version, schema, description = entry.split('|')
+        result.append({'name': name,
+                       'version': version,
+                       'schema': schema,
+                       'description': description})
+
+    return result
+
+def generate_report(packages):
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    DOC_URL = 'https://www.postgresql.org/docs/10/upgrading.html'
```

Comment:
We should ideally link to internal upgrade steps (currently available only in 8.0 Beta docs), and we don't know a correct link that will be ready after GA. How to handle this? Any idea, @pirat89 @bocekm?

Reply:
@hhorak just post a patch once you have it, or have an RedHat KB article you can with some text and update the link in that article once you have the right link

Reply:
As I've mentioned in another conversation, the URL is now changed to the KB article. This should be fixed as well.